conference/src/test/java/com/kyriba/conference/api/HallApiTest.java

```diff
@@ -0,0 +1,166 @@
+package com.kyriba.conference.api;
+
+
+import com.kyriba.conference.api.dto.HallRequest;
+import com.kyriba.conference.api.dto.HallResponse;
+import com.kyriba.conference.domain.Hall;
+import com.kyriba.conference.service.HallServiceImpl;
+import io.restassured.builder.RequestSpecBuilder;
+import io.restassured.specification.RequestSpecification;
+import org.apache.http.HttpStatus;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.restdocs.JUnitRestDocumentation;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.util.List;
+
+import static io.restassured.RestAssured.given;
+import static java.util.Arrays.asList;
+import static junit.framework.TestCase.assertEquals;
+import static junit.framework.TestCase.assertNotNull;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doReturn;
+import static org.springframework.http.MediaType.APPLICATION_JSON_UTF8_VALUE;
+import static org.springframework.restdocs.restassured3.RestAssuredRestDocumentation.document;
+import static org.springframework.restdocs.restassured3.RestAssuredRestDocumentation.documentationConfiguration;
+
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT)
+@AutoConfigureMockMvc
```

Comment:
Annotation is not needed.

Reply:
the test is failed in gradle (gradle clean build) without this annotation. It works only if I define 'RANDOM_PORT' and `@LocalServerPort int port;`, but how should I deal with 'DEFINED_PORT'?